HACKING.md

```diff
@@ -42,3 +44,37 @@ For a quick compile and debug cycle, we recommend iterating code using thorough
 unit tests before going through the full workflow described above.
 
 [1]: https://wiki.qemu.org/Documentation/Networking#User_Networking_.28SLIRP.29
+
+## Containers
+
+*osbuild-composer* and *osbuild-composer-worker* can be run using Docker
+containers. Building and starting containers is generally faster than building
+RPMs and installing them in a VM, so this method is more convenient for
+developing and testing changes quickly.
+
+Each service (*composer* and *worker*) requires a configuration file and a set
+of certificates. Use the [`tools/gen-certs.sh`](./tools/gen-certs.sh) script to
+generate the certificates (using the test OpenSSL config file):
+
+    ./tools/gen-certs.sh ./test/data/x509/openssl.cnf ./containers/config  ./containers/config/ca
+
+The services also require a config file (each) which they expect to be in the
+same directory. The following test files can be copied into it:
+
+    cp ./test/data/composer/osbuild-composer.toml ./test/data/composer/osbuild-worker.toml ./containers/config/
```

Comment:
They could but then we'd be generating certs inside a directory checked into git and that feels a bit messy.  The way the guide is written now, the `config` directory can be placed anywhere and the `docker-composer.yml` volumes updated for anyone who wants to work outside the source tree.

Now that I think about it, I'd prefer to make that path configurable (through `distribution/.env`) and add an instruction at the start that it can be changed to be anywhere (and keep it in `containers/config` by default).

Reply:
ah, got it, makes sense